row_cache.cc

```diff
@@ -47,7 +47,7 @@ using namespace cache;
 flat_mutation_reader
 row_cache::create_underlying_reader(read_context& ctx, mutation_source& src, const dht::partition_range& pr) {
     ctx.on_underlying_created();
-    return src.make_reader(_schema, pr, ctx.slice(), ctx.pc(), ctx.trace_state(), streamed_mutation::forwarding::yes);
+    return src.make_reader(_schema, pr, ctx.slice(), ctx.pc(), ctx.trace_state(), streamed_mutation::forwarding::yes, mutation_reader::forwarding::yes);
```

Comment:
When `pr` is singular we will create an underlying sstable reader that doesn't support `mutation_reader::forwarding::yes`. Not a problem currently, because we are just silently dropping unsupported parameters left and right and you will just get a `std::bad_function_call` if you actually try to fast forward the reader. But maybe we should propagate a forwarding param here that is set to `mutation_reader::forwarding::no` when `pr` is singular.

Reply:
@tgrabiec what do you think?